### PR TITLE
Adds some more best practices to the GraphQL docs

### PR DESCRIPTION
**Pull request by orta**
2 come from our Exchange work, and another comes from my Peril work that ideally I would have done had I had those ideas earlier.

### CODE REVIEW

File: best-practices/graphql-schema-design.md [hunk 1]
```diff
@@ -166,3 +166,114 @@ microservices.
...
+
+This ensures that:
+
+- You can never end up in a state where `is_pickup` is true, but there is address metadata available
+- You can safely extend `Shipping` with a new type (like a digital work with a url/email)
+- Clients need to specify and be aware of the objects they want to handle when making queries
+
+## Mutation Responses as Unions
+
+The GraphQL community still hasn't consolidate on how to handle errors in mutations. We're currently thinking that
```
Comment (damassi) [inline best-practices/graphql-schema-design.md]: consolidated

File: best-practices/graphql-schema-design.md [hunk 2]
```diff
@@ -166,3 +166,114 @@ microservices.
...
+there are two types of issues: exceptions and errors. An exception is something unexpected, and this shows up in
+root of a response under `errors`.
+
+For errors that you expect to happen, then they can be handled as a union.
```
Comment (alloy) [inline best-practices/graphql-schema-design.md]: > Errors that you expect to happen can be modeled as a union.

File: best-practices/graphql-schema-design.md [hunk 3]
```diff
@@ -166,3 +166,114 @@ microservices.
...
+```graphql
+type ConsignmentSubmission {
+  title: String!
+  location: String!
+  category: String!
+
+  # This is still optional
+  signatureExplanation: String
+}
+```
```
Comment (alloy) [inline best-practices/graphql-schema-design.md]: I’m not sure I understand how this is used. Does the client have to resend all the data that was previously already submitted as a draft?

Comment (ashkan18) [reply best-practices/graphql-schema-design.md]: do these end up being passed to the same mutation?

**Comment by orta** [comment]
@alloy this got an update

File: best-practices/graphql-schema-design.md [hunk 4]
```diff
@@ -166,3 +166,164 @@ microservices.
...
+  id: ID
+  title: String
+  location: String
+  category: String
+
+  # This is optional
+  signatureExplanation: String
+}
+
+# A fully-validated Consignment, with non-null versions of it's inputs
```
Comment (alloy) [inline best-practices/graphql-schema-design.md]: > its

**Comment by alloy** [comment]
One tiny error, but otherwise LGTM. Feel free to merge once fixed 👍

**Comment by orta** [comment]
cool, merge on green

File: best-practices/graphql-schema-design.md (hunk 3 above)
Comment (orta) [reply best-practices/graphql-schema-design.md]: I rewrote the section with examples, and should negate the need for these qs

**Pull request merged**